bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderWatcherHandler.java

```diff
@@ -0,0 +1,291 @@
+                            "Connetction timeout can't be negative");
+                    return;
+                }
+
+                if (config.secureMode.equals("NONE")) {
+                    ftp = new FTPClient();
+                } else {
+                    FTPSClient ftps;
+                    if (config.secureMode.equals("IMPLICIT")) {
+                        ftps = new FTPSClient(true);
+                    } else if (config.secureMode.equals("EXPLICIT")) {
+                        ftps = new FTPSClient(false);
+                    } else {
+                        ftps = null;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Unsupported secure mode");
+                        return;
+                    }
+                    ftp = ftps;
+                }
+
+                reply = 0;
+                ftp.setListHiddenFiles(config.listHidden);
+                ftp.setConnectTimeout(config.connectionTimeout * 1000);
+                // Uncomment to see FTP response
+                // ftp.addProtocolCommandListener(new PrintCommandListener(new PrintWriter(System.out), true));
+
+                try {
+                    if (config.ftpPort > 0) {
+                        ftp.connect(config.ftpAddress, config.ftpPort);
+                    } else {
+                        ftp.connect(config.ftpAddress);
+                    }
+                    reply = ftp.getReplyCode();
+
+                    if (!FTPReply.isPositiveCompletion(reply)) {
+                        ftp.disconnect();
+                        logger.debug("FTP server refused connection.");
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "FTP server refused connection.");
+                        return;
+                    }
+                } catch (IOException e) {
+                    if (ftp.isConnected()) {
+                        try {
+                            ftp.disconnect();
+                        } catch (IOException e2) {
+                            logger.debug("Can't disconect: ", e2);
+                        }
+                    }
+                    logger.debug("Can't conect: ", e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    return;
+                }
+                __main: try {
+                    if (!ftp.login(config.ftpUsername, config.ftpPassword)) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ftp.getReplyString());
+                        ftp.logout();
+                        break __main;
+                    }
+                    updateStatus(ThingStatus.ONLINE);
+                    return;
+                }
+
+                catch (FTPConnectionClosedException e) {
```

Comment:
```suggestion
                catch (FTPConnectionClosedException | IOException e) {
```

You can catch both exception in one clause since your reaction is the same anyway.